### DISCUSSION TITLE
AMS run's out of JVM Heap Memory

### DISCUSSION THREAD

**Discussion by mstepuch** [ant-media/Ant-Media-Server#5521] [Q&A] [answered]
Hello 
I have AMS community installed on VPS server. 4 cores 12Gb RAM, 150 Gb disk space.
My JVM Heap Memory i set to 6Gb

I run three IP cameras connected as LiveStreams to WebRTCApp and I retransmit them to YouTube.
They run 24/7.

Last days when I switched from one to three steams, server constantly runs out of JVM Heap Memory.
When I restart it, memory usage grows and within about three hours server is unresponsive ang stops transmitting to YT.

Any idea what I can do?? Thank You in advice

**Comment by USAMAWIZARD** [comment]
@mstepuch Which version of Ant Media are you using which protocols are you using for publishing Live IP Camera streams to AMS.

RTSP or RTMP ??

**Comment by mstepuch** [comment]
I was pushing streams from cameras using RTMP with adress like this
rtmp://adress-of-my-AMS-server.com/WebRTCApp/mystream
These are Dahua IP cams

Now i'am testing conection to cameras usin IPCamera option in WebRTCApp (ONVIF)
But as I see nothing changed.

I forgot to mention I'am running AMS on Ubuntu Server. Not containered

**Reply by USAMAWIZARD** [reply]
I think its related to a memory leak issue that have been fixed recently.

**Comment by mstepuch** [comment]
Thank You for willing to help.
I reinstalled AMS with newest version Last week.
I am running Community Edition 2.6.4 20230827_1333, I dont think there is any newer version

**Reply by USAMAWIZARD** [reply]
I mean its not released yet there is a PR for that fix , If you are planning to get Enterprise  I can provide you a snapshot version of the fix

**Reply by mstepuch** [reply, marked as answer]
Thank You but i'll stay with community ed. 
Strange, but when I changed video options of my cams now it seams to be OK.
Memory usage stopped at about 1,5 Gb
One of options which I disabled was Dahua Samrt Codec.
Other was bitrate from CBR to VBR.
